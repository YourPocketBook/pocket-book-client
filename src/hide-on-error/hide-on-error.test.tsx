import { render } from "@testing-library/react";
import React from "react";
import { HideOnError } from ".";

function TestControl({ fail }: { fail: boolean }) {
  if (fail) {
    throw Error();
  }
  return <div className="test" />;
}

describe("Hide On Error", () => {
  it("shows a successful control", () => {
    const { container } = render(
      <HideOnError>
        <TestControl fail={false} />
      </HideOnError>
    );

    expect(container).toMatchSnapshot();
  });

  it("hides a failing control", () => {
    const { container } = render(
      <HideOnError>
        <TestControl fail={true} />
      </HideOnError>
    );

    expect(container).toMatchSnapshot();
  });
});
