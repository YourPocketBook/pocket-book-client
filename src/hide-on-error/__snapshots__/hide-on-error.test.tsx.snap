// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hide On Error hides a failing control 1`] = `<div />`;

exports[`Hide On Error shows a successful control 1`] = `
<div>
  <div
    class="test"
  />
</div>
`;
